src/GrpcBootstrap.php

```diff
@@ -32,32 +32,55 @@
 namespace Google\GAX;
 
 use Google\Auth\ApplicationDefaultCredentials;
+use Google\Auth\CredentialsLoader;
 use Grpc\ChannelCredentials;
 
 /**
  * Class that provides helpful utility functions for bootstrapping a gRPC client.
  */
 class GrpcBootstrap
 {
+    private $authCredentials;
+
     /**
-     * Provides a default instance of GrpcBootstrap.
+     * Accept an optional keyFile argument, which can be used to load credentials
+     * instead of using ApplicationDefaultCredentials
      */
-    public static function defaultInstance()
+    public function __construct($scopes, $options = [])
     {
-        return new GrpcBootstrap();
+        if (array_key_exists('keyFile', $options) && !is_null($options['keyFile'])) {
+            # Get credentials from the keyFile provided
+            $jsonData = json_decode(file_get_contents($options['keyFile']), true);
+            $this->authCredentials = CredentialsLoader::makeCredentials($scopes, $jsonData);
+        } else {
+            # Get the Application Default Credentials
+            $this->authCredentials = ApplicationDefaultCredentials::getCredentials($scopes);
```

Comment:
I would probably use  getADCCredentials() here so it is overridable in the unit test. or we can remove getADCCredentials since it is no longer used.